compiler/rustc_next_trait_solver/src/solve/trait_goals.rs

```diff
@@ -189,6 +189,20 @@ where
             debug_assert!(ecx.opaque_type_is_rigid(opaque_ty.def_id));
         }
 
+        if let ty::CoroutineWitness(def_id, _) = goal.predicate.self_ty().kind() {
```

Comment:
I should pull this out into a helper, b/c I think I need to also apply this hack to copy/clone. I think those are it tho.

Reply:
https://crater-reports.s3.amazonaws.com/pr-133502-2/try%23da0a4799770027c4c578bf48cdb390acfde09023/reg/kitsune2_test_utils-0.1.3/log.txt